athena-docdb/src/main/java/com/amazonaws/athena/connectors/docdb/DocDBMetadataHandler.java

```diff
@@ -160,15 +168,58 @@ public ListSchemasResponse doListSchemaNames(BlockAllocator blockAllocator, List
     public ListTablesResponse doListTables(BlockAllocator blockAllocator, ListTablesRequest request)
     {
         MongoClient client = getOrCreateConn(request);
-        List<TableName> tables = new ArrayList<>();
 
         try (MongoCursor<String> itr = client.getDatabase(request.getSchemaName()).listCollectionNames().iterator()) {
+            List<TableName> tables = new ArrayList<>();
             while (itr.hasNext()) {
                 tables.add(new TableName(request.getSchemaName(), itr.next()));
             }
 
             return new ListTablesResponse(request.getCatalogName(), tables, null);
         }
+        catch (MongoCommandException mongoCommandException) {
+            //do this in failed case instead of replace method in case API changes on doc db.
+            logger.warn("Exception on listCollectionNames on Mongo JAVA client, trying with mongo command line.", mongoCommandException);
+            List<TableName> tableNames = doListTablesWithCommand(client, request);
```

Comment:
is there a way to detect the underlying data-source, and perform this proactively? instead of falling back on an exception?

Reply:
db.hostInfo return this information but this required an Admin permission on DB, which not all data user will have. Therefore, doing this mechanism to not break or introduce any additional permission